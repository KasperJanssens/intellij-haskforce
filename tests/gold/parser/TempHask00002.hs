{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE QuasiQuotes       #-}
{-# LANGUAGE TemplateHaskell   #-}
module TempHask00002 where

[undefined|
something here "hello world"
|]

