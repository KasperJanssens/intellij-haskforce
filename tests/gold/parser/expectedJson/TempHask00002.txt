[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":9,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":9,"srcSpanEndLine":9,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":22}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":8},"srcInfoPoints":[]},"TempHask00002"]},null,null]},[{"LanguagePragma":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":32}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":31,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":14},"srcInfoPoints":[]},"FlexibleInstances"]}]]},{"LanguagePragma":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":32}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":14},"srcInfoPoints":[]},"QuasiQuotes"]}]]},{"LanguagePragma":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":32}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":14},"srcInfoPoints":[]},"TemplateHaskell"]}]]}],[],[{"SpliceDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":6,"srcSpanEndLine":8,"srcSpanStartColumn":1},"srcInfoPoints":[]},{"QuasiQuote":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":3,"srcSpanStartLine":6,"srcSpanEndLine":8,"srcSpanStartColumn":1},"srcInfoPoints":[]},"undefined","\nsomething here \"hello world\"\n"]}]}]]},[]]