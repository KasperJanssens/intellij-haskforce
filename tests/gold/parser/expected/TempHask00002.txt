Haskell File
  HaskellPpragmaImpl(PPRAGMA)
    PsiElement(HaskellTokenType.{-#)('{-#')
    PsiElement(HaskellTokenType.PRAGMA)(' LANGUAGE FlexibleInstances ')
    PsiElement(HaskellTokenType.#-})('#-}')
  HaskellPpragmaImpl(PPRAGMA)
    PsiElement(HaskellTokenType.{-#)('{-#')
    PsiElement(HaskellTokenType.PRAGMA)(' LANGUAGE QuasiQuotes       ')
    PsiElement(HaskellTokenType.#-})('#-}')
  HaskellPpragmaImpl(PPRAGMA)
    PsiElement(HaskellTokenType.{-#)('{-#')
    PsiElement(HaskellTokenType.PRAGMA)(' LANGUAGE TemplateHaskell   ')
    PsiElement(HaskellTokenType.#-})('#-}')
  PsiElement(HaskellTokenType.module)('module')
  HaskellQconidImpl(QCONID)
    PsiElement(HaskellTokenType.conid)('TempHask00002')
  PsiElement(HaskellTokenType.where)('where')
  HaskellBodyImpl(BODY)
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('undefined')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('something')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('here')
    HaskellPstringtokenImpl(PSTRINGTOKEN)
      PsiElement(HaskellTokenType.")('"')
      PsiElement(HaskellTokenType.STRINGTOKEN)('hello world')
      PsiElement(HaskellTokenType.")('"')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    PsiElement(HaskellTokenType.integertoken)('7')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('d')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    PsiElement(HaskellTokenType.integertoken)('7')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('t')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conid)('Int')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('p')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    PsiElement(HaskellTokenType.integertoken)('7')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.$)('$')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('deriveStuff')
    PsiElement(HaskellTokenType.')(''')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('f')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('m')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.$)('$')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('x')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.'')('''')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conid)('T')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('f')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('n')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.\)('\')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.[)('[')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('haskell')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('y')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.|)('|')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.])(']')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.->)('->')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('y')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('n')